sklearn/utils/validation.py

```diff
@@ -505,6 +519,10 @@ def check_array(array, accept_sparse=False, accept_large_sparse=True,
         # DataFrame.sparse only supports `to_coo`
         array = array.sparse.to_coo()
 
+    if has_pd_interger_array:
+        # convert dataframe with Integer extension arrays with floats
+        array = array.astype(None)
```

Comment:
I suspect that `IntegerArray.astype(none)` returning a float64 ndarray isn't deliberate, right @jorisvandenbossche?.

Reply:
I don't think that is deliberate, indeed (I would say that's a bug .. although it seems consistent with numpy where `np.dtype(None)` is float64).